Makefile

```diff
@@ -270,13 +270,8 @@ go-generate: $(BIN)/mockgen $(BIN)/enumer
 	@echo "updating copyright headers"
 	@$(MAKE) --no-print-directory copyright
 
-lint: $(BIN)/golint fmt
-	@echo "running linter"
-	@lintFail=0; for file in $(sort $(LINT_SRC)); do \
-		$(BIN)/golint "$$file"; \
-		if [ $$? -eq 1 ]; then lintFail=1; fi; \
-	done; \
-	if [ $$lintFail -eq 1 ]; then exit 1; fi;
+lint: $(BIN)/revive
+	$(BIN)/revive -config revive.toml -exclude './canary/...' -exclude './vendor/...' -formatter unix ./... | sort
```

Comment:
strangely it also doesn't exclude vendor by default.  maybe if I set my go flags to include -mod=vendor...?

meh.  at least you can _tell it to ignore it_.